statsmodels/multivariate/factor.py

```diff
@@ -0,0 +1,245 @@
+        self.n_comp = matrix_rank(R)
+        if self.n_factor > self.n_comp:
+            raise ValueError('n_factor must be smaller or equal to the rank'
+                             ' of endog! %d > %d' %
+                             (self.n_factor, self.n_comp))
+        if n_max_iter <= 0:
+            raise ValueError('n_max_iter must be larger than 0! %d < 0' %
+                             (n_max_iter))
+        if tolerance <= 0 or tolerance > 0.01:
+            raise ValueError('tolerance must be larger than 0 and smaller than'
+                             ' 0.01! Got %f instead' % (tolerance))
+        #  Initial communality estimation
+        if SMC:
+            c = 1 - 1 / np.diag(inv(R))
+            self.SMC = np.array(c)
+        else:
+            c = np.ones([1, len(R)])
+
+        # Iterative communality estimation
+        eigenvals = None
+        for i in range(n_max_iter):
+            # Get eigenvalues/eigenvectors of R with diag replaced by
+            # communality
+            for j in range(len(R)):
+                R[j, j] = c[j]
+            L, V = eig(R)
+            c_last = np.array(c)
+            ind = np.argsort(L)
+            ind = ind[::-1]
+            L = L[ind]
+            n_pos = (L > 0).sum()
+            V = V[:, ind]
+            eigenvals = np.array(L)
+
+            # Select eigenvectors with positive eigenvalues
+            n = np.min([n_pos, self.n_factor])
+            sL = np.diag(np.sqrt(L[:n]))
+            V = V[:, :n]
+
+            # Calculate new loadings and communality
+            A = V.dot(sL)
+            c = np.power(A, 2).sum(axis=1)
+            if norm(c_last - c) < tolerance:
+                break
+        self.eigenvals = eigenvals
+        self.communality = c
+        # Perform rotation of the loadings
+        self.loadings_no_rot = np.array(A)
+        if rotation in ['varimax', 'quartimax', 'biquartimax', 'equamax',
+                        'parsimax', 'parsimony', 'biquartimin']:
+            A, T = rotate_factors(A, rotation)
+        elif rotation == 'oblimin':
+            A, T = rotate_factors(A, 'quartimin')
+        elif rotation == 'promax':
+            A, T = promax(A)
+
+        if rotation is not None:  # Rotated
+            c = np.power(A, 2).sum(axis=1)
+        self.loadings = A
+        self.rotation = rotation
+        return FactorResults(self)
+
+    def plot_scree(self, ncomp=None):
+        """
+        Plot of the ordered eigenvalues and variance explained for the loadings
+
+        Parameters
+        ----------
+        ncomp : int, optional
+            Number of loadings to include in the plot.  If None, will
+            included the same as the number of maximum possible loadings
+
+        Returns
+        -------
+        fig : figure
+            Handle to the figure
+        """
+        return plot_scree(self.eigenvals, self.n_comp, ncomp)
+
+    def plot_loadings(self, loading_pairs=None, plot_prerotated=False):
```

Comment:
this should also be a results method because it requires that `fit` has been called